### PR TITLE
Replace custom exception handler with whoops

### LINKED ISSUE #89: Add content negotiation to exception handler
Right now the exception handler always generates JSON. It would be really nice to use [whoops](https://github.com/filp/whoops) in Spark for handling exceptions when HTML output is useful. Along the lines of the `FormattedResponder` it would be nice if the exception handler could be content negotiated.

### PR DESCRIPTION
**Pull request by shadowhand**
Having a content negotiated exception handler provides more robust
debugging capabilities in multiple contexts. [Whoops][1] has the added
benefit of being capable of logging.

By having a separate class for the preferred handlers the user can
customize which handlers will be used easily.

[1]: https://github.com/filp/whoops

Fixes #89

### CODE REVIEW

File: src/Handler/ExceptionHandlerPreferences.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            'application/javascript' => JsonResponseHandler::class,
+            'application/json' => JsonResponseHandler::class,
+            'applicaiton/ld+json' => JsonResponseHandler::class,
+            'application/vnd.api+json' => JsonResponseHandler::class,
+            'application/vnd.geo+json' => JsonResponseHandler::class,
+            'application/xml' => XmlResponseHandler::class,
+            'application/atom+xml' => XmlResponseHandler::class,
+            'application/rss+xml' => XmlResponseHandler::class,
+            'text/plain' => PlainTextHandler::class,
+        ];
```
Comment (shadowhand) [inline src/Handler/ExceptionHandlerPreferences.php]: Somehow this line isn't covered. Need to figure it out with Travis.

File: src/Handler/ExceptionHandler.php [hunk 2]
```diff
@@ -4,47 +4,50 @@
...
+    /**
+     * @var Whoops
+     */
+    private $whoops;
 
-        return $copy;
+    /**
+     * @param ExceptionHandlerPreferences $preferences
+     * @param NegotiatorInterface $negotiator
+     * @param ResolverInterface $resolver
```
Comment (elazar) [inline src/Handler/ExceptionHandler.php]: Need a `@param` for the `Whoops` parameter.

File: src/Handler/ExceptionHandler.php [hunk 3]
```diff
@@ -55,14 +58,16 @@ public function withRoot($root)
      * @return ResponseInterface
      */
     public function __invoke(
-        RequestInterface  $request,
+        ServerRequestInterface $request,
```
Comment (elazar) [inline src/Handler/ExceptionHandler.php]: `@param` tag type should be updated.

File: src/Handler/ExceptionHandlerPreferences.php (hunk 1 above)
Comment (elazar) [inline src/Handler/ExceptionHandlerPreferences.php]: Probably just a coveralls snafu. I wouldn't worry about it too much.

**Comment by elazar** [comment]
Other than those nitpicks, :+1: :shipit:

File: src/Handler/ExceptionHandler.php (hunk 3 above)
Comment (shadowhand) [inline src/Handler/ExceptionHandler.php]: Fixed.

File: src/Handler/ExceptionHandler.php (hunk 2 above)
Comment (shadowhand) [inline src/Handler/ExceptionHandler.php]: Fixed.

File: src/Handler/ExceptionHandlerPreferences.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            'application/javascript' => JsonResponseHandler::class,
+            'application/json' => JsonResponseHandler::class,
+            'applicaiton/ld+json' => JsonResponseHandler::class,
+            'application/vnd.api+json' => JsonResponseHandler::class,
+            'application/vnd.geo+json' => JsonResponseHandler::class,
+            'application/xml' => XmlResponseHandler::class,
+            'application/atom+xml' => XmlResponseHandler::class,
+            'application/rss+xml' => XmlResponseHandler::class,
+            'text/plain' => PlainTextHandler::class,
+        ]; // @codeCoverageIgnore
```
Comment (shadowhand) [inline src/Handler/ExceptionHandlerPreferences.php]: :smiling_imp:

**Comment by ameech** [comment]
Nice!

**Pull request merged**